### ISSUE TITLE
Usage with targets outside render

### ISSUE THREAD

**Issue by 0m15** [react-spring/react-use-gesture#47]
Hi, 
this is not a real issue, just a quick tip for people who needs the listener to be added to a target outside of the component render, for example, `window`:

```
  const bind = useGesture(() => {})
  const cb = bind()
  window.addEventListener("mousedown", cb.onMouseDown, { passive: true })
  window.addEventListener("touchstart", cb.onTouchStart, { passive: true })
```

This works fine.

**Comment by drcmda** [comment]
Next version will have api support with automatic cleanup for that (@dbismut made it), so that you can do something like this:

```jsx
const bind = useGesture(..., config: { target })
useEffect(bind, [target])
```

**Comment by dbismut** [comment]
Hi @0m15, we've just released a beta of v5.0.0 (`react-use-gesture@5.0.0-beta.3`) that does what @drcmda mentioned. The proper api is:

```jsx
const bind = useGesture({ onDrag: ... }, { domTarget: window })
React.useEffect(bind, [bind]) // binds to window, and removes listeners on unmount
```
